apps/admin-gui/src/app/vos/pages/vo-detail-page/vo-settings/vo-settings-member-organizations/vo-settings-member-organizations.component.ts

```diff
@@ -0,0 +1,105 @@
+import { ReloadEntityDetailService } from '../../../../../core/services/common/reload-entity-detail.service';
+import { UniversalRemoveItemsDialogComponent } from '@perun-web-apps/perun/dialogs';
+import { TranslateService } from '@ngx-translate/core';
+
+@Component({
+  selector: 'app-vo-settings-member-organizations',
+  templateUrl: './vo-settings-member-organizations.component.html',
+  styleUrls: ['./vo-settings-member-organizations.component.scss'],
+})
+export class VoSettingsMemberOrganizationsComponent implements OnInit {
+  loading = false;
+  voId: number;
+  voSelection: SelectionModel<Vo> = new SelectionModel<Vo>(false, []);
+  displayedColumns = ['checkbox', 'id', 'shortName', 'name'];
+  filterValue = '';
+  auth = false;
+  memberVos: Vo[] = [];
+
+  constructor(
+    private dialog: MatDialog,
+    private vosService: VosManagerService,
+    private entityStorage: EntityStorageService,
+    private authResolver: GuiAuthResolver,
+    private reloadDetailService: ReloadEntityDetailService,
+    private notificator: NotificatorService,
+    private translator: TranslateService
+  ) {}
+
+  ngOnInit(): void {
+    this.auth = this.authResolver.isPerunAdmin();
+    this.voId = this.entityStorage.getEntity().id;
+    this.refresh();
+  }
+
+  refresh(): void {
+    this.loading = true;
+    this.vosService.getEnrichedVoById(this.voId).subscribe(
+      (enrichedVo) => {
+        this.voId = enrichedVo.vo.id;
+        this.memberVos = enrichedVo.memberVos;
+        this.voSelection.clear();
+        this.loading = false;
+      },
+      () => (this.loading = false)
+    );
+  }
+
+  addMemberOrganization(): void {
+    const config = getDefaultDialogConfig();
+    config.width = '750px';
+
+    this.dialog
+      .open(AddMemberOrganizationDialogComponent, config)
+      .afterClosed()
+      .subscribe((result) => {
+        if (result) {
+          this.reloadDetailService.reloadEntityDetail();
+          this.refresh();
+        }
+      });
+  }
+
+  removeMemberVos() {
+    this.vosService.removeMemberVo(this.voId, this.voSelection.selected[0].id).subscribe(() => {
+      this.notificator.showSuccess(
+        this.translator.instant(
+          'VO_DETAIL.SETTINGS.MEMBER_ORGANIZATIONS.REMOVE_MEMBER_ORGANIZATION.TITLE'
+        ) as string
+      );
+    });
+  }
+
+  removeMemberOrganization() {
+    const config = getDefaultDialogConfig();
+    config.width = '550px';
+    config.data = {
+      items: this.voSelection.selected.map((vo) => vo.name),
+      title: 'VO_DETAIL.SETTINGS.MEMBER_ORGANIZATIONS.REMOVE_MEMBER_ORGANIZATION.TITLE',
+      alert: 'VO_DETAIL.SETTINGS.MEMBER_ORGANIZATIONS.REMOVE_MEMBER_ORGANIZATION.WARNING',
+      theme: 'vo-theme',
+    };
+
+    this.dialog
+      .open(UniversalRemoveItemsDialogComponent, config)
+      .afterClosed()
+      .subscribe((result) => {
+        if (result) {
+          this.removeMemberVos();
+          this.reloadDetailService.reloadEntityDetail();
```

Comment:
reload entity detail doesnt wait for remove member vos to end so the entity is reloaded with old data